openmls/src/ciphersuite/crypto/evercrypt_provider.rs

```diff
@@ -0,0 +1,187 @@
+
+use std::convert::TryFrom;
+
+use evercrypt::prelude::*;
+use log::error;
+
+use crate::ciphersuite::{errors::CryptoError, AeadType, HashType, SignatureScheme};
+
+impl TryFrom<SignatureScheme> for SignatureMode {
+    type Error = &'static str;
+    #[inline(always)]
+    fn try_from(signature_scheme: SignatureScheme) -> Result<Self, Self::Error> {
+        match signature_scheme {
+            SignatureScheme::ED25519 => Ok(SignatureMode::Ed25519),
+            SignatureScheme::ECDSA_SECP256R1_SHA256 => Ok(SignatureMode::P256),
+            SignatureScheme::ED448 => Err("SignatureScheme ed448 is not supported."),
+            SignatureScheme::ECDSA_SECP521R1_SHA512 => {
+                Err("SignatureScheme ecdsa_secp521r1 is not supported.")
+            }
+        }
+    }
+}
+
+impl TryFrom<SignatureScheme> for DigestMode {
+    type Error = &'static str;
+    #[inline(always)]
+    fn try_from(signature_scheme: SignatureScheme) -> Result<Self, Self::Error> {
+        match signature_scheme {
+            // The digest mode for ed25519 is not really used
+            SignatureScheme::ED25519 => Ok(DigestMode::Sha256),
+            SignatureScheme::ECDSA_SECP256R1_SHA256 => Ok(DigestMode::Sha256),
+            SignatureScheme::ED448 => Err("SignatureScheme ed448 is not supported."),
+            SignatureScheme::ECDSA_SECP521R1_SHA512 => {
+                Err("SignatureScheme ecdsa_secp521r1 is not supported.")
+            }
+        }
+    }
+}
+#[inline(always)]
+fn hash_from_algorithm(hash_type: HashType) -> Result<DigestMode, CryptoError> {
+    Ok(match hash_type {
+        HashType::Sha1 => DigestMode::Sha1,
+        HashType::Sha2_224 => DigestMode::Sha224,
+        HashType::Sha2_256 => DigestMode::Sha256,
+        HashType::Sha2_384 => DigestMode::Sha384,
+        HashType::Sha2_512 => DigestMode::Sha512,
+        HashType::Sha3_224 => DigestMode::Sha3_224,
+        HashType::Sha3_256 => DigestMode::Sha3_256,
+        HashType::Sha3_384 => DigestMode::Sha3_384,
+        HashType::Sha3_512 => DigestMode::Sha3_512,
+        _ => return Err(CryptoError::UnsupportedHashAlgorithm),
+    })
+}
+
+#[inline(always)]
+fn aead_from_algorithm(alg: AeadType) -> Result<AeadMode, CryptoError> {
+    match alg {
+        AeadType::Aes128Gcm => Ok(AeadMode::Aes128Gcm),
+        AeadType::Aes256Gcm => Ok(AeadMode::Aes256Gcm),
+        AeadType::ChaCha20Poly1305 => Ok(AeadMode::Chacha20Poly1305),
+        AeadType::HpkeExport => Err(CryptoError::UnsupportedAeadAlgorithm),
+    }
+}
+
+#[inline(always)]
+fn kdf_from_hash(hash_type: HashType) -> Result<HmacMode, CryptoError> {
+    Ok(match hash_type {
+        HashType::Sha1 => HmacMode::Sha1,
+        HashType::Sha2_256 => HmacMode::Sha256,
+        HashType::Sha2_384 => HmacMode::Sha384,
+        HashType::Sha2_512 => HmacMode::Sha512,
+        _ => return Err(CryptoError::UnsupportedKdf),
+    })
+}
+
+pub(crate) fn support(signature_scheme: SignatureScheme) -> Result<(), CryptoError> {
```

Comment:
Nitpicking, but `support` sounds a bit strange. Maybe `supports`, `has_support`, `check_support`?

Reply:
This function will go away anyway, but sure I can call it `support`